test/IDE/coloring_class_restriction.swift

```diff
@@ -0,0 +1,5 @@
+// RUN: %target-swift-ide-test -syntax-coloring -source-filename %s | %FileCheck %s
```

Comment:
Because I want the `class` be the end of the file because the assertion hit in https://bugs.swift.org/browse/SR-8378 caused by out of range `CharSourceRange`.
I wanted to make sure this test case doesn't hit it. I know closing tag `</type>` may ensures that, but still.

Reply:
That's fair, do add a comment saying that, and especially mentioning that nothing should be added after the `protocol P : class` line.

Reply:
Added a commit.